netsim/modules/stp.py

```diff
@@ -0,0 +1,71 @@
+#
+# STP transformation module
+#
+from box import Box
+
+from ..augment import devices
+from ..utils import log
+
+from . import _Module
+
+class STP(_Module):
+
+  # Check for device support when pvrst is required
+  def node_pre_transform(self, node: Box, topology: Box) -> None:
+
+    if not topology.stp.get('pvrst',False):
```

Comment:
This will create topology.stp if it doesn't exist. Are you counting on global stp.enable being there?